rs/bitcoin/adapter/BUILD.bazel

```diff
@@ -109,10 +109,10 @@ rust_test_suite(
         "@bitcoin_adapter_mainnet_headers//file",
         "@bitcoin_adapter_testnet_blocks//file",
         "@bitcoin_adapter_testnet_headers//file",
-        "@bitcoin_core//:bitcoind",
+        "@bitcoin_core_x86_64-linux-gnu//:bitcoind",
     ],
     env = {
-        "BITCOIN_CORE_PATH": "$(rootpath @bitcoin_core//:bitcoind)",
+        "BITCOIN_CORE_PATH": "$(rootpath @bitcoin_core_x86_64-linux-gnu//:bitcoind)",
```

Comment:
There must a better way of doing it.  I don't think here we should even mention the platform

cc: @nmattia  @basvandijk

Reply:
https://dfinity.slack.com/archives/CL7Q2RXUM/p1732785051529109

Reply:
sure thing! we do this for [jemalloc](https://github.com/dfinity/ic/tree/master/third_party/jemalloc) and [lmdb](https://github.com/dfinity/ic/tree/master/third_party/lmdb), among others. Let me know if you need a hand!

Reply:
I could do it by reverting #1603 : it was removed according to [this](https://github.com/dfinity/ic/pull/1491#discussion_r1762760330) conversation.

Reply:
Note that the bitcoind version becomes downgraded by reverting the commit: 27.0 => 25.0. Are we fine with that?

Reply:
Building from sources doesn't work on macOS: https://github.com/dfinity/ic/actions/runs/12089105211/job/33713847474?pr=2890 - how should we proceed (I'd very much like to have bitcoin tests for PocketIC on macOS, but don't have time to debug building bitcoind from sources on macOS)? Would it be fine to download pre-built artifacts as I did originally after all?

Reply:
Followed the approach from this PR: https://github.com/dfinity/ic/pull/2996